pennylane/devices/default_qubit_autograd.py

```diff
@@ -116,6 +115,15 @@ class DefaultQubitAutograd(DefaultQubit):
     _roll = staticmethod(np.roll)
     _stack = staticmethod(np.stack)
 
+    @staticmethod
+    def _asarray(array, dtype=None):
+        res = np.asarray(array, dtype=dtype)
+
+        if res.dtype is np.dtype("O"):
+            return np.asarray(np.hstack(array).flatten(), dtype=dtype)

```

Comment:
The reason behind using both `np.hstack` and `.flatten()` here to support flattening multdim arrays, right? And the `asarray` is simply to get the `dtype` in as well. Any reason for not using `np.hstack(array).flatten().astype(dtype)` instead (although I would say both work)?

Reply:
Oh I like your suggestion better
```suggestion
            return np.hstack(array).flatten().astype(dtype)
```